backend/app/experiments/models.py

```diff
@@ -0,0 +1,736 @@
+    )
+    user_id: Mapped[int] = mapped_column(
+        Integer, ForeignKey("users.user_id"), nullable=False
+    )
+    workspace_id: Mapped[int] = mapped_column(
+        Integer, ForeignKey("workspace.workspace_id"), nullable=False
+    )
+
+    # Description
+    name: Mapped[str] = mapped_column(String(length=150), nullable=False)
+    description: Mapped[str] = mapped_column(String(length=500), nullable=False)
+    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
+
+    # Assignments config
+    sticky_assignment: Mapped[bool] = mapped_column(
+        Boolean, nullable=False, default=False
+    )
+    auto_fail: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
+    auto_fail_value: Mapped[int] = mapped_column(Integer, nullable=True)
+    auto_fail_unit: Mapped[AutoFailUnitType] = mapped_column(
+        Enum(AutoFailUnitType), nullable=True
+    )
+
+    # Experiment config
+    exp_type: Mapped[str] = mapped_column(String(length=50), nullable=False)
+    prior_type: Mapped[str] = mapped_column(String(length=50), nullable=False)
+    reward_type: Mapped[str] = mapped_column(String(length=50), nullable=False)
+
+    # State variables
+    created_datetime_utc: Mapped[datetime] = mapped_column(
+        DateTime(timezone=True), nullable=False
+    )
+    n_trials: Mapped[int] = mapped_column(Integer, nullable=False)
+    last_trial_datetime_utc: Mapped[datetime] = mapped_column(
+        DateTime(timezone=True), nullable=True
+    )
+
+    # Relationships
+    arms: Mapped[list["ArmDB"]] = relationship(
+        "ArmDB", back_populates="experiment", lazy="joined"
+    )
+    draws: Mapped[list["DrawDB"]] = relationship(
+        "DrawDB",
+        back_populates="experiment",
+        primaryjoin="ExperimentDB.experiment_id==DrawDB.experiment_id",
+        lazy="joined",
+    )
+    clients: Mapped[list["ClientDB"]] = relationship(
+        "ClientDB",
+        back_populates="experiment",
+        lazy="joined",
+    )
+    contexts: Mapped[Optional[list["ContextDB"]]] = relationship(
+        "ContextDB",
+        back_populates="experiment",
+        lazy="joined",
+        primaryjoin="and_(ExperimentDB.experiment_id==ContextDB.experiment_id,"
+        + "ExperimentDB.exp_type=='cmab')",
+    )
+
+    def __repr__(self) -> str:
+        """
+        String representation of the model
+        """
+        return f"<Experiment(name={self.name}, type={self.exp_type})>"
+
+    @property
+    def has_contexts(self) -> bool:
+        """Check if this experiment type supports contexts."""
+        return self.exp_type == "cmab"
+
+    @property
+    def context_list(self) -> list["ContextDB"] | list:
+        """Get contexts, returning empty list if not applicable."""
+        return self.contexts if self.has_contexts and self.contexts is not None else []
+
+    def to_dict(self) -> dict:
+        """
+        Convert the ORM object to a dictionary.
+        """
+        return {
```

Comment:
Should we be returning `clients` here as well?

Reply:
Will this get super heavy once we have a tonne of draws / clients? 
I guess it's fine for now but would be not explicitly load those unless needed